File: 백준/Silver/15656. N과 M （7）/N과 M （7）.swift
var nums = readLine()!.split(separator: " ").map {Int(String($0))!}
var numbers = readLine()!.split(separator: " ").map {Int(String($0))!}
numbers.sort()
var selected = [String]()
var answer = ""

func DFS(start: Int) {
    if selected.count == nums[1] {
        answer += selected.joined(separator: " ") + "\n"
        return
    }
    for num in 0..<nums[0] {
        selected.append(String(numbers[num]))
        DFS(start: start+1)
        selected.removeLast()
    }
}

DFS(start:0)
print(answer)

File: 백준/Silver/15656. N과 M （7）/N과 M （7）.py
n, m = map(int, input().split())
nums = list(map(int, input().split()))
nums.sort()

lst = []

def DFS():
    if len(lst) == m:
        print(" ".join(map(str, lst)))
        return
    for i in range(0, n):
        # if nums[i] not in lst:
        lst.append(nums[i])
        DFS()
        lst.pop()

DFS()